srml/balances/src/lib.rs

```diff
@@ -1165,34 +1189,49 @@ impl<T: Trait<I>, I: Instance> rstd::fmt::Debug for TakeFees<T, I> {
 	}
 }
 
-use primitives::traits::{DispatchError, SaturatedConversion};
-use primitives::transaction_validity::ValidTransaction;
-use primitives::weights::Weight;
-
 impl<T: Trait<I>, I: Instance + Clone + Eq> SignedExtension for TakeFees<T, I> {
 	type AccountId = T::AccountId;
 	type AdditionalSigned = ();
 	fn additional_signed(&self) -> rstd::result::Result<(), &'static str> { Ok(()) }
 
+	fn pre_dispatch(
```

Comment:
why introduce this? by default it just calls `validate`.

Reply:
Because I am proposing that maybe the logic in them is slightly different: In validate:
- Can just call `ensure_can_withdraw`.
- Omit the imbalance stuff.

If not reasonable, then it should be reverted and only use `validate()`.

Reply:
fair enough, but then maybe there should be an auxilliary to deduplicate the code.